### PR TITLE
[Feature] Add GraphStorm docs in the rst format

### PR DESCRIPTION
**Pull request by zhjwy9343**
*Issue #, if available:*

*Description of changes:*
Add a new docs/ folder, which have current documents in the rst format

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: docs/source/configuration/configuration-partition.rst [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+* `partition_graph.py <https://github.com/awslabs/graphstorm/blob/main/tools/partition_graph.py>`_: for Node/Edge Classification/Regress task graph partition.
+* `partition_graph_lp.py <https://github.com/awslabs/graphstorm/blob/main/tools/partition_graph_lp.py>`_: for Link Prediction task graph partition.
+
+`partition_graph.py <https://github.com/awslabs/graphstorm/blob/main/tools/partition_graph.py>`_ arguments
+---------------------------------------------------------------------------------------------------------------
+
+- **--dataset**: (**Required**) the graph dataset name defined for the saved DGL graph file.
+- **--filepath**: (**Required**) the file path of the saved DGL graph file.
+- **--target_ntype**: the node type for making prediction, required for node classification/regression tasks. This argument is associated with the node type having labels. Current GraphStorm supports **one** predict node type only.
```
Comment (classicsong) [inline docs/source/configuration/configuration-partition.rst:16]:
```suggestion
- **--target-ntype**: the node type for making prediction, required for node classification/regression tasks. This argument is associated with the node type having labels. Current GraphStorm supports **one** predict node type only.
```

File: docs/source/configuration/configuration-partition.rst [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+- **--ntype_task**: the node type task to perform. Only support ``classification`` and ``regression`` so far. Default is ``classification``.
```
Comment (classicsong) [inline docs/source/configuration/configuration-partition.rst:17]: Change _ to - for all the arguments.

File: docs/source/configuration/configuration-partition.rst [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+- **--retain_original_features**: boolean value to control if use the original features generated by dataset, e.g., embeddings of paper abstracts. If set to ``true``, will keep the original features; otherwise we will use the tokenized text for using BERT models to generate embeddings.
+- **--num_parts**: (**Required**) integer value that specifies partitions the DGL graph to be split. Remember this number because we will need to set it in the model training step.
+- **--output**: (**Required**) the folder path that the partitioned DGL graph will be saved.
+
+`partition_graph_lp.py <https://github.com/awslabs/graphstorm/blob/main/tools/partition_graph_lp.py>`_ arguments
+------------------------------------------------------------------------------------------------------------------------------------
+- **--dataset**: (**Required**) the graph name defined for the saved DGL graph file.
+- **--filepath**: (**Required**) the file path of the saved DGL graph file.
+- **--target_etypes**: (**Required**) the canonical edge type for making prediction.  GraphStorm supports **one** predict edge type only. The format is ``src_ntype,etype,dst_ntype``, e.g., `"author,write,paper"`.
```
Comment (classicsong) [inline docs/source/configuration/configuration-partition.rst:39]:
```suggestion
- **--target-etypes**: (**Required**) the canonical edge type for making prediction.  GraphStorm supports **one** predict edge type only. The format is ``src_ntype,etype,dst_ntype``, e.g., `"author,write,paper"`.
```

Same for the rest.

File: docs/source/scale/distributed.rst [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+
+.. note:: If possible, use **private IP addresses**, insteand of public IP addresses. Public IP addresses may have additional port constraints, which cause communication issues.
+    
+Put this file into container's ``/data/`` folder.
+
+Check port 2222
+................
+The GraphStorm Docker container uses port ``2222`` to **ssh** to containers running on other machines without passwords. Please make sure all host instances do not use this port.
+
+Users also need to make sure the port ``2222`` is open for **ssh** commands.
```
Comment (classicsong) [inline docs/source/scale/distributed.rst:61]: For distributed training, we also need to make sure ports under 65536 is Open for DistDGL to use.

File: docs/source/tutorials/quick-start.rst [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+
+Download  and Partition OGB-arxiv Data
+--------------------------------------
+First run the below command.
+
+.. code-block:: bash
+
+    python3 /graphstorm/tools/partition_graph.py --dataset ogbn-arxiv \
+                                                 --filepath /tmp/ogbn-arxiv-nc/ \
+                                                 --num_parts 1 \
```
Comment (classicsong) [inline docs/source/tutorials/quick-start.rst:27]:
```suggestion
                                                 --num-parts 1 \
```

File: docs/source/advanced/own-models.rst [hunk 6]
```diff
@@ -0,0 +1,398 @@
...
+                                                                                    # while another is the computed node representations, which can be saved.
+
+The ``create_optimizer()`` function is for users to define their own optimizer. You can put the optimizer definition from the training flow inside the model, like the code below
+
+.. code-block:: python
+
+    def create_optimizer(self, lr=0.001):
+        return torch.optim.Adam(self.parameters(), lr=lr)
+
+There are other optional functions in the `GSgnnNodeModelBase <https://github.com/awslabs/graphstorm/blob/main/python/graphstorm/model/node_gnn.py#L76>`_ class, including ``restore_model(self, restore_model_path)`` and ``save_model(self, model_path)``, which are used to restore and save models. If you want to save or restore models, implement these two functions too.
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/advanced/own-models.rst:169]: But it is runnable even without implementation of the two functions. So they are actually optional.

File: docs/source/advanced/own-models.rst [hunk 7]
```diff
@@ -0,0 +1,398 @@
...
+
+.. code-block:: python
+
+    train_data = GSgnnNodeTrainData(config.graph_name,
+                                    config.part_config,
+                                    train_ntypes=config.target_ntype,
+                                    node_feat_field=node_feat_fields,
+                                    label_field=config.label_field)
+
+Arguments of this class include the partition configuration JSON file path, which are the outputs of the :ref:`Step 1 <step-1>`. The ``graph_name`` can be found in the JSON file.
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/advanced/own-models.rst:203]: this is a customized configuration. so its value cannot be automatically extracted. Users need to provide this argument.

File: docs/source/advanced/own-models.rst [hunk 8]
```diff
@@ -0,0 +1,398 @@
...
+GraphStorm has a set of parameters that control the various perspectives of the model training and inference process. You can find the details of these parameters in the GraphStorm :ref:`Training and Inference Configurations <configurations-run>`. These parameters could be either passed as input arguments or set in a YAML format file. Below is an example of the YAML file.
+
+.. code-block:: yaml
+
+    ---
+    version: 1.0
+    gsf:
+    basic:
+        backend: gloo
+        ip_config: ip_list.txt
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/advanced/own-models.rst:332]: removed.

File: docs/source/advanced/own-models.rst [hunk 9]
```diff
@@ -0,0 +1,398 @@
...
+
+    python3 ~/dgl/tools/launch.py \
+            --workspace /hgt_nc \
+            --part-config /hgt_nc/acm_data.json \
+            --ip-config ip_list.txt \
+            --num-trainers 4 \
+            --num-servers 1 \
+            --num-samplers 0 \
+            --ssh-port 2222 \
+            "python3 hgt_nc.py --yaml-config-file acm_nc.yaml \
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/advanced/own-models.rst:391]: Yes, tested in the PR #177 and changed.

File: docs/source/advanced/own-models.rst [hunk 10]
```diff
@@ -0,0 +1,398 @@
...
+
+            h[ntype] = F.gelu(n_embed)
+
+        for i in range(self.num_layers):
+            h = self.gcs[i](blocks[i], h)
+
+        return h[self.target_ntype].argmax(dim=1), h[self.target_ntype]             # return two values: one is the predict results, 
+                                                                                    # while another is the computed node representations, which can be saved.
+
+The ``create_optimizer()`` function is for users to define their own optimizer. You can put the optimizer definition from the training flow inside the model, like the code below
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/advanced/own-models.rst:162]: Removed this sentence.

File: docs/source/configuration/configuration-gconstruction.rst [hunk 11]
```diff
@@ -0,0 +1,138 @@
...
+
+* ``task_type``: (**Required**) specifies the task defined on the nodes or edges. Currently, its value can be ``classification``, ``regression`` and ``link_prediction``.
+* ``label_col``: specifies the column name in the input file that contains the label. This has to be specified for ``classification`` and ``regression`` tasks. ``label_col`` is used as the label name.
+* ``split_pct``: specifies how to split the data into training/validation/test. This is optional. If it's not specified, the data is split into 80% for training 10% for validation and 10% for testing. The pipeline constructs three additional vectors indicating the training/validation/test masks. For ``classification`` and ``regression`` tasks, the names of the mask tensors are ``train_mask``, ``val_mask`` and ``test_mask``.
+
+.. _input-format:
+
+Input formats
+..............
+Currently, the graph construction pipeline supports two input formats: ``Parquet``, ``HDF5``, and ``JSON``.
```
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/configuration/configuration-gconstruction.rst:69]: changed to three.

File: docs/source/configuration/configuration-partition.rst (hunk 1 above)
Comment (zhjwy9343) [reply docs/source/configuration/configuration-partition.rst:16]: All arguments of the graph_partition.py and graph_partition_lp.py are still using "_", not "-". 

So, I am not sure if we need to change those arguments before v0.1 release.

File: docs/source/configuration/configuration-partition.rst (hunk 3 above)
Comment (zhjwy9343) [reply docs/source/configuration/configuration-partition.rst]: all changed.

File: docs/source/tutorials/quick-start.rst (hunk 5 above)
Comment (zhjwy9343) [reply docs/source/tutorials/quick-start.rst]: changed.

File: docs/source/advanced/own-models.rst (hunk 6 above)
[the remark this answers is not in the document]
Comment (zhjwy9343) [reply docs/source/advanced/own-models.rst]: add the two functions in the docs

**Review by classicsong — APPROVED** [review]

**Pull request merged**